dor/providers/process_basic_image.py

```diff
@@ -0,0 +1,212 @@
+    temp_file = tempfile.NamedTemporaryFile(mode='w', suffix=".tiff")
+    with temp_file:
+        if source_tech_metadata.compressed or source_tech_metadata.rotated:
+            compressible_file_path = Path(temp_file.name)
+            make_intermediate_file(new_source_file_path, compressible_file_path)
+        else:
+            compressible_file_path = new_source_file_path
+
+        try:
+            generate_service_variant(compressible_file_path, service_file_path)
+        except ServiceImageProcessingError:
+            return False
+
+    service_event = create_preservation_event("generate service derivative", collection_manager_email)
+    service_event_xml = PreservationEventSerializer(service_event).serialize()
+    service_event_metadata_file_info = service_image_file_info.metadata(UseFunction.event, "text/xml+premis")
+    (output_path / identifier / service_event_metadata_file_info.path).write_text(service_event_xml)
+
+    try:
+        service_tech_metadata = get_technical_metadata(service_file_path)
+    except TechnicalMetadataError as error:
+        return False
+
+    service_tech_meta_file_info = service_image_file_info.metadata(
+        UseFunction.technical, service_tech_metadata.metadata_mimetype.value
+    )
+    (output_path / identifier / service_tech_meta_file_info.path).write_text(service_tech_metadata.metadata)
+
+    descriptor_file_path = output_path / identifier / "descriptor" / f"{identifier}.file_set.mets2.xml"
+
+    resource = PackageResource(
+        id=file_set_identifier.uuid,
+        type="File Set",
+        alternate_identifier=AlternateIdentifier(id=file_set_identifier.alternate_identifier, type="DLXS"),
+        events=[],
+        metadata_files=[
+            convert_metadata_file_info_to_file_metadata(tech_meta_file_info),
+            convert_metadata_file_info_to_file_metadata(service_tech_meta_file_info),
+            convert_metadata_file_info_to_file_metadata(source_event_metadata_file_info),
+            convert_metadata_file_info_to_file_metadata(service_event_metadata_file_info)
+        ],
+        data_files=[
+            FileMetadata(
+                id=image_file_info.xmlid,
+                use=flatten_use(*image_file_info.uses),
+                mdid=tech_meta_file_info.xmlid,
+                ref=FileReference(
+                    locref=image_file_info.locref,
+                    mimetype=image_file_info.mimetype
+                )
+            ),
+            FileMetadata(
+                id=service_image_file_info.xmlid,
+                use=flatten_use(*service_image_file_info.uses),
+                groupid=image_file_info.xmlid,
+                mdid=service_tech_meta_file_info.xmlid,
+                ref=FileReference(
+                    locref=service_image_file_info.locref,
+                    mimetype=service_image_file_info.mimetype
+                )
+            ),
+        ]
+    )
+
+    create_file_set_descriptor_file(resource, descriptor_file_path)
+
```

Comment:
Perhaps extracting the steps into function calls will make it more readable …. just to many lines, lol